server/boot/routes.js

```diff
@@ -101,12 +100,9 @@ module.exports = function(app) {
         req.body.password !== req.body.confirmation) {
       return res.sendStatus(400, new Error('Passwords do not match'));
     }
-
-    User.findById(req.accessToken.userId, function(err, user) {
+      User.setPassword(req.accessToken.userId, req.body.password,
```

Comment:
We need to verify that `req.accessToken` is scoped to allow the scope assigned to `User.setPassword`, that was the reason for https://github.com/strongloop/loopback/pull/3350.

I am proposing to rework the app to invoke the built-in REST endpoint `POST /api/users/reset-password` (see https://github.com/strongloop/loopback/pull/3350). Similarly as with 7bdf18a2b (change password), we should add an after-remote hook to send back a redirect response.

BTW when writing a web application (as opposed to an API server), it's a common practice for POST requests to return a redirect instead of an HTML body. That way the browser loads another page via GET and when the user ask the browser to reload the page, only the GET page is reloaded. Had the POST request returned an HTML body, the browser would ask the user whether they want to re-submit the data via POST.